src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -120,6 +125,51 @@
     ProgressMeter progressMeter;
 
     /**
+     * Return the list of GATKCommandLinePluginDescriptors to be used for this tool.
+     * Uses the read filter plugin.
+     */
+    @Override
+    protected List<? extends CommandLinePluginDescriptor<?>> getPluginDescriptors() {
+        return Collections.singletonList(new GATKReadFilterPluginDescriptor(getDefaultReadFilters()));
+    }
+
+    /**
+     * Returns the default list of CommandLineReadFilters that are used for this tool. The filters returned
+     * by this method are subject to selective enabling/disabling by the user via the command line. The
+     * default implementation returns an empty list. Subclasses can override to provide alternative filters.
+     *
+     * Note: this method is called before command line parsing begins, and thus before a SAMFileHeader is
+     * available through {link #getHeaderForReads}.
+     *
+     * @return List of individual filters to be applied for this tool.
+     */
+    public List<ReadFilter> getDefaultReadFilters() {
+        return Collections.singletonList(new WellformedReadFilter());
+    }
+
+    /**
+     * Returns a read filter (simple or composite) that can be applied to reads.
+     * The default implementation combines the default read filters for this tool (returned by
+     * {@link org.broadinstitute.hellbender.engine.GATKTool#getDefaultReadFilters} with any read filter command
+     * line arguments specified by the user; wraps each filter in the resulting list with a CountingReadFilter;
+     * and returns a single composite filter resulting from the list by and'ing them together.
+     *
+     * Default tool implementation of {@link #traverse()} calls this method once before iterating
+     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter
+     * objects is strongly discouraged.
+     *
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter}
+     * composition methods.
+     */
+    public CountingReadFilter makeReadFilter(){
+        final GATKReadFilterPluginDescriptor readFilterPlugin =
+                commandLineParser.getPluginDescriptor(GATKReadFilterPluginDescriptor.class);
+        return getHeaderForReads() == null ?
```

Comment:
Hm. I don't think we have a clear story on that. It looks like we have some code that assumes that if we have reads, we have a header (and sequence dictionary), and some code that guards. But I don't see anyplace where we enforce that a header is present (and our use of setHeaderStrict implies that we require one). The spec says the header is optional. It's pretty easy to get an NPE with SplitReads by trying a samfile with no header. I propose we settle on a policy, and I'll make a separate ticket to audit/enforce whatever we choose. Let me know what you think.